### PR TITLE
Add docstrings.

### PR DESCRIPTION
**Pull request by scottdangelo**
When you are happy, please merge. I'll continue with a new patch set.

### CODE REVIEW

File: ?
Comment (stevemar) [inline ?:1]: these can go back 4 spaces and be lined up with the first "

Comment (stevemar) [inline ?:1]: new line between description "common data in dict2 ..." and the parameter bits

Comment (stevemar) [inline ?:1]: these can be bumped back too

Comment (markstur) [inline ?:1]: :param message: <desc>
or
:param str message: <desc>
but definitetly colon before param and after message

Comment (markstur) [reply ?:1]: :returns:
(starts with colon)

Comment (markstur) [inline ?:1]: I'm pretty sure no space after """ although I never liked that look. Maybe this is allowed(?)

Comment (markstur) [inline ?:1]: s/rtype:/:rtype:/

Comment (markstur) [inline ?:1]: Here s/type/rtype/ and all the before/after colon stuff mentioned above

**Review by markstur — COMMENTED** [review]
Some comments on the syntax

**Pull request merged**